mobius-core/src/test/java/com/spotify/mobius/TestConsumer.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * -\-\-
+ * Mobius
+ * --
+ * Copyright (c) 2017-2018 Spotify AB
+ * --
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * -/-/-
+ */
+package com.spotify.mobius;
+
+import com.spotify.mobius.functions.Consumer;
+import java.util.Collections;
+import java.util.List;
+import org.assertj.core.util.Lists;
+
+class TestConsumer<T> implements Consumer<T> {
```

Comment:
didn't we already have something similar to this? Perhaps we can use it or replace it with this one?

Reply:
There is `RecordingConsumer`, but that would lead to a cyclic dependency as it's an external utility in `mobius-test`. Otherwise, I can't find anything, so I think this is fine.

Reply:
mobius-test has a compileOnly dependency on mobius-core precisely for this reason, so it should work? I'm pretty sure we already use things from mobius-test in mobius-core

Reply:
Right, yes. That is an interesting setup, not quite sure how it works, but it's good. :) I think I'll add a method to read the values from the `RecordingConsumer`, to open up for using AssertJ and Hamcrest matchers more flexibly.

Reply:
Actually, maybe using that consumer/changing its API should be a separate PR, because it might be worth spending a little time on in itself.

Reply:
sure - let's try to get that done before doing the next release.